force-app/main/default/lwc/donationHistoryTable/donationHistoryTable.js

```diff
@@ -5,13 +5,13 @@ import PAYMENT_FIELD from '@salesforce/schema/DataImport__c.Payment_Method__c';
 import donationHistoryDatatableAriaLabel from '@salesforce/label/c.donationHistoryDatatableAriaLabel';
 import RD2_ScheduleVisualizerColumnDate from '@salesforce/label/c.RD2_ScheduleVisualizerColumnDate';
 import getDonationHistory from '@salesforce/apex/DonationHistoryController.getDonationHistory';
+import getTotalNumberOfRecords from '@salesforce/apex/DonationHistoryController.getTotalNumberOfRecords';
```

Comment:
I had another suggestion in [PR #6813](https://github.com/SalesforceFoundation/NPSP/pull/6813/files#r787308992) on how we should be trying to retrieve additional information from the back-end that I think is worth implementing here too. Lets include the `totalNumberOfRecords` as part of the response returned by `getDonationHistory`.

Instead of returning a `List<DonationHistoryView>` from `getDonationHistory`, we'd return a `DonationHistoryAppView` that looks like this:

```
public class DonationHistoryAppView {
  @AuraEnabled List<DonationHistoryView> donations;
  @AuraEnabled Integer totalNumberOfRecords;
}
```

Reply:
response changed!